### PR TITLE
Compatibility with ingress and self signed cert

### PR DESCRIPTION
**Pull request by gabrieloandco**
Hi, we are setting up teleport in our environment and we needed to make these changes on the helm chart to make them configurable with our ingress controller and self signed certifcates. We use sslpassthrough with nginx to get it working. However we could'nt use our self certificate securely because teleport sshproxy is not compatible with Cloudflare proxy (https://github.com/gravitational/teleport/issues/7962)

### CODE REVIEW

**Comment by russjones** [comment]
@webvictim Can you review?

File: examples/chart/teleport-cluster/values.yaml [hunk 1]
```diff
@@ -28,6 +28,33 @@ acmeEmail: ""
...
+  k8s_port: "3026"
+  sshtun_port: "3024"
+
+ingress:
+  enabled: false
+  # Used to create Ingress record (should used with service.type: ClusterIP).
+  hosts:
+    - ""
+  # - teleport.domain.com
+  # The ingress path. Useful to host teleport on a subpath, such as `/teleport`.
```
Comment (programmerq) [inline examples/chart/teleport-cluster/values.yaml:45]: Is this tested functionality? I'm not sure that tsh would be able to find teleport if it's anywhere other than `/`.

**Review by programmerq — COMMENTED** [review]
This version of the changes has an all-in or all-out approach to the Ingress.

Many kubernetes ingress implementations will make it difficult for the ports that need to be handled as tcp services: 3023, 3024, 3026.

In my opinion, opting in to an Ingress option should affect only the proxy web api port (443) and not the others.

**Comment by gabrieloandco** [comment]
> This version of the changes has an all-in or all-out approach to the Ingress.
> 
> Many kubernetes ingress implementations will make it difficult for the ports that need to be handled as tcp services: 3023, 3024, 3026.
> 
> In my opinion, opting in to an Ingress option should affect only the proxy web api port (443) and not the others.

1- With an Nginx Ingress Controller this is very possible and I have it deployed that way and any Ingress with Layer 4 compatibility (which I'm pretty sure that the Cloud Providers are compatible). For the Nginx Ingress Controller is just a value to add to the helm chart to allow TCP ports. 

2-tsh doesn't work well with custom ports nor domains right now so it will look for all these ports in the same domain. Then the approach proposed in the PR works since in the same domain you should have all the ports that tsh needs to use, and they have to be the default ones because again custom ports don't work well right now. 

3-For security reasons you want your cluster to be only available from an ingress and not a LB. Ingresa have many features for monitoring and security that LB just don't have. Anyway if you still want to use LB you can just set the ingress off and the LB on.

File: examples/chart/teleport-cluster/values.yaml (hunk 1 above)
Comment (gabrieloandco) [reply examples/chart/teleport-cluster/values.yaml:45]: If you define the complete path as a proxy in tsh it shouldn't have any issue.

Comment (webvictim) [reply examples/chart/teleport-cluster/values.yaml:45]: Teleport doesn't support the usage of a subpath for `tsh` or other tooling - the service must be available at the root of the proxy URL.

**Comment by zmb3** [comment]
Hey @webvictim - I think there was a recent change to support bring your own secret. Does that mean we can close this one?

**Comment by webvictim** [comment]
This doesn't do exactly the same thing as bringing your own TLS cert, but I agree that this is idle enough now that it can be closed. I am probably going to look at independently adding `Ingress` support to the `teleport-cluster` chart soon.

**Pull request closed without merging**